datacompy/core.py

```diff
@@ -229,6 +230,14 @@ def _dataframe_merge(self, ignore_spaces):
         else:
             params = {"on": self.join_columns}
 
+        if ignore_spaces:
+            self.df1[self.join_columns] = self.df1[self.join_columns].apply(
+                lambda x: x.str.strip() if x.dtype.kind == "O" else x
```

Comment:
Hmm, so `apply` is not the fastest...I guess you did this due to the element dtype check. How come you're doing that one? Maybe we could do it vectorized? All the `str` methods work as vectorized methods.

Reply:
Yeah just for the check really. I don’t want to strip something which isn’t
a string as it will throw an error.

Reply:
I guess I can loop through the list check the dtype and then just vertorize right? Assuming that is what you were thinking?

Reply:
@jborchma Made a change to move away from `apply`. Does this feel better. Agree apply is known to be slow, and this could lead to some savings.